src/Microsoft.OData.Core/Json/JsonLightInstanceAnnotationWriter.cs

```diff
@@ -84,27 +84,98 @@ internal JsonLightInstanceAnnotationWriter(ODataJsonLightValueSerializer valueSe
         /// <param name="ignoreFilter">Whether to ignore the filter in settings.</param>
         /// <param name="propertyName">The name of the property this instance annotation applies to</param>
         internal void WriteInstanceAnnotations(
-            IEnumerable<ODataInstanceAnnotation> instanceAnnotations,
+            ICollection<ODataInstanceAnnotation> instanceAnnotations,
             InstanceAnnotationWriteTracker tracker,
             bool ignoreFilter = false,
             string propertyName = null)
         {
             Debug.Assert(instanceAnnotations != null, "instanceAnnotations should not be null if we called this");
             Debug.Assert(tracker != null, "tracker should not be null if we called this");
 
+            // this method runs in a hot path hence the optimizations
+
+            if (instanceAnnotations.Count == 0)
+            {
+                return;
+            }
+
+#if NETSTANDARD2_1_OR_GREATER

```

Comment:
@habbes Where would .NetStandard2.1 or greater apply? On which frameworks will this evaluate to true? We support Net45, NetStandard1.1 and NetStandard2.0 in ODL so I'd like to be sure this is not dead code and that it's testable... Otherwise maybe this should go into the 8.x major release
Another question, would `#if NETSTANDARD2_1_OR_GREATER` evaluate to true where the target framework is Net5?

Reply:
I changed this to `#if NETSTANDARD2_1`, the `NETSTANDARD2_1_OR_GREATER` was a mistake from autocomplete.

netstandard2.1 is supported by net fx 4.8, net core 3.1, .net 5.

But you may have a point, since we don't seem to explicitly target those platforms. Let me confirm.

Reply:
You were right. I've looked at all the generated binaries, and none of them use the code in the #if block. So it seems to be dead code unless we explicitly target netstandard21.